File: flutter_app/lib/Historico.dart
import 'package:json_annotation/json_annotation.dart';
part 'Historico.g.dart';

@JsonSerializable()
class Historico {
  Historico(this.dataPregao, this.valorAbertura, this.variacaoDiaAnterior, this.variacaoInicioPeriodo);

  final DateTime dataPregao;
  final double valorAbertura;
  final double variacaoDiaAnterior;
  final double variacaoInicioPeriodo;

  factory Historico.fromJson(Map<String, dynamic> json) => _$HistoricoFromJson(json);
  Map<String, dynamic> toJson() => _$HistoricoToJson(this);
}

File: VariacaoAtivoBackend/VariacaoAtivoBackend/Historico.cs
using System;

namespace VariacaoAtivoBackend
{
    public class Historico
    {        
        public Historico(DateTime dataPregao, decimal valorAbertura)
        {
            DataPregao = dataPregao;
            ValorAbertura = valorAbertura;
        }

        public DateTime DataPregao { get; protected set; }
        public decimal ValorAbertura { get; protected set; }
        public decimal VariacaoDiaAnterior { get; set; }
        public decimal VariacaoInicioPeriodo { get; set; }
    }
}
